### PR TITLE
feat: Support "Keyword Args" sections for G-style

### LINKED ISSUE #88: Support "Keyword Arguments" sections in Google-style docstrings
**Is your feature request related to a problem? Please describe.**
Keyword sections is useful to document the possible contents of `**kwargs`. Without it, users must write keyword arguments documentation in the `Arguments` section, under the `**kwargs` parameter, using plain Markdown, therefore not benefiting from the table template.

**Describe the solution you'd like**
Support keyword arguments section in the google style parser.

**Describe alternatives you've considered**
As mentioned previously, an alternative is to document kwargs in the Arguments section, but this is a bit tedious to do it right.

**Additional context**
Only the Google-style parser is concerned by this feature request. Other issues will be needed for the other parsers, if needed.
Originally asked on mkdocstrings: https://github.com/pawamoy/mkdocstrings/issues/205

### PR DESCRIPTION
**Pull request by HacKanCuBa**
Add support for "Keyword Args" and "Keyword Arguments" sections for
Google-style docstrings.

Closes #88.

### CODE REVIEW

**Comment by HacKanCuBa** [comment]
Let me know if the changes and tests make sense, it seemed way too simple so I may be totally wrong.

**Comment by HacKanCuBa** [comment]
On a side note, running `make check` produces:
```
✗ Checking code quality (1)
✓ Type-checking
✓ Building documentation
✓ Checking dependencies
```

So I tried running `flakehell lint src/pytkdocs` but got an exception: `AttributeError: 'Namespace' object has no attribute 'extended_default_ignore'` so I'm not sure what's going on there.

**Comment by pawamoy** [comment]
> Let me know if the changes and tests make sense, it seemed way too simple so I may be totally wrong.

All fine here, it was supposed to be easy :slightly_smiling_face:

File: src/pytkdocs/parsers/docstrings/google.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
     "arguments:": Section.Type.PARAMETERS,
     "params:": Section.Type.PARAMETERS,
     "parameters:": Section.Type.PARAMETERS,
+    "keyword args:": Section.Type.PARAMETERS,
+    "keyword arguments:": Section.Type.PARAMETERS,
```
Comment (pawamoy) [inline src/pytkdocs/parsers/docstrings/google.py:13]: I think that for the sake of rendering with a different title than "Parameters" in mkdocstrings, we will need a new section type. Probably `Section.Type.KEYWORD_PARAMETERS`. It means you'll also have to add an key-value pair in the parser `section_reader` attribute. We can still reuse the same `read_parameters_section` though :slightly_smiling_face:

**Review by pawamoy — CHANGES_REQUESTED** [review]
That's good! We just need to differentiate an "Arguments" section from a "Keyword Arguments" one :slightly_smiling_face:

File: src/pytkdocs/parsers/docstrings/google.py (hunk 1 above)
Comment (pawamoy) [reply src/pytkdocs/parsers/docstrings/google.py:13]: Ah, the method returns hardcodes a `Section.Type.PARAMETERS` though. Maybe accept a new arg in the method, and set partials in `section_reader`. Or refactor the method into three (one that does almost everything while the other two just create the actual section with the right type).

**Comment by pawamoy** [comment]
Also, don't worry about the quality check, it is supposed to fail (we allow failures).

File: src/pytkdocs/parsers/docstrings/google.py (hunk 1 above)
Comment (HacKanCuBa) [reply src/pytkdocs/parsers/docstrings/google.py:13]: I was thinking about this. I checked how the other parsers were done and i.e. for rst the args are the same: `PARAM_NAMES = frozenset(("param", "parameter", "arg", "argument", "key", "keyword"))`, so I thought that for g-style the same would apply.

However, having a different proper Section type would certainly be better. And I think that unhardcoding is the way to go so I'm going to refactor the `read_parameters_section` method into three.

Comment (HacKanCuBa) [reply src/pytkdocs/parsers/docstrings/google.py]: Changes pushed

**Comment by HacKanCuBa** [comment]
> Also, don't worry about the quality check, it is supposed to fail (we allow failures).

Nice, thanks. In any case, if it's something I can fix related to this PR, I'm happy to fix it! I couldn't find an output that tells me why that check fails

File: src/pytkdocs/parsers/docstrings/google.py [hunk 2]
```diff
@@ -265,12 +268,46 @@ def read_parameters_section(self, lines: List[str], start_index: int) -> Tuple[O
...
+
+        Returns:
+            A tuple containing a `Section` (or `None`) and the index at which to continue parsing.
+        """
+        parameters, i = self._parse_parameters_section(lines, start_index)
+
+        if parameters:
+            return Section(Section.Type.KEYWORD_ARGS, parameters), i
+
+        self.error(f"Empty parameters section at line {start_index}")
```
Comment (pawamoy) [inline src/pytkdocs/parsers/docstrings/google.py:308]:
```suggestion
        self.error(f"Empty keyword arguments section at line {start_index}")
```

Comment (HacKanCuBa) [reply src/pytkdocs/parsers/docstrings/google.py:308]: good catch!

**Comment by pawamoy** [comment]
Thanks again!

**Pull request merged**